### ISSUE TITLE
`make error!` should cause error immediately

### ISSUE THREAD

**Issue by bitbegin** [red/red#3592]
**Describe the bug**
In Rebol:

```
>> x: func [a][if a = 1 [make error! "333"] 3]
>> x 1
** User Error: 333
** Near: make error! "333"
>>
```

In Red: 

```
>> x: func [a][if a = 1 [make error! "333"] 3]
== func [a][if a = 1 [make error! "333"] 3]
>> x 1
== 3
>> 
```

`make error!` should cause error immediately, otherwise there are no way to throw this error!.

**To Reproduce**
```
>> x: func [a][if a = 1 [make error! "333"] 3]
== func [a][if a = 1 [make error! "333"] 3]
>> x 1
== 3
>> 
```

**Expected behavior**
```
>> x: func [a][if a = 1 [make error! "333"] 3]
>> x 1
** User Error: 333
** Near: make error! "333"
>>
```

**Screenshots**

**Platform version(please complete the following information):**
```
Red 0.6.3 for Windows built 13-Nov-2018/23:49:48+08:00 commit #7404492
```

**Comment by meijeru** [comment]
To throw the error, `do` it. See also `source cause-error`

**Comment by endo64** [comment]
This was a design decision IIRC. @greggirwin ?

**Comment by greggirwin** [comment]
This is an intentional design change. (typing at the same time Endo).

**Comment by greggirwin** [comment]
The console evaluates errors if it sees them, but errors are not "hot" during normal evaluation.
```
>> make error! 1
*** Throw Error: return or exit not in function
*** Where: ??? 

>> reduce [make error! 1]
== [make error! [
    code: 1
    type: 'throw
    id: 'return
    arg1: none
    arg2: none
    arg3: none
    near: none...
```

**Comment by bitbegin** [comment]
I don't think we can just close this issue, as there are some reasons we should fix this design.

1: now `make error!` only give an error type value, we just can return it, and if it's last value in `try` block, it'll be caught:

```
>> x: func [a][if a = 1 [return make error! "333"] 3]
== func [a][if a = 1 [return make error! "333"] 3]
>> print try [x 1]
*** User Error: "333"
*** Where: ???
```

But, if it's not last value, it'll be discarded:

```
>> x: func [a][if a = 1 [return make error! "333"] 3]
== func [a][if a = 1 [return make error! "333"] 3]
>> print try [x 1 2]
2
>> 
```

2: `make error!` not be designed as an exception? This will be strange behavior, and not uniform with other places

3: if `make error!` be designed as an exception, but just give an error! type value, there should exist `throw` method to raise it. (As `throw` be used in `catch/throw` block, maybe lead into other keyword like `raise` )

**Comment by greggirwin** [comment]
Reopened, so Nenad can comment definitively.

1) is only true in the console. As shown in my `reduce` example, error values are not discarded, they are just not hot when created with `make`.

2) I believe it is NOT designed to be an exception, but a way to create error values, which may be used in various ways. Listing some other places that are inconsistent with this would be good.

3) That's what `cause-error` is for.

**Comment by qtxie** [comment]
1. `make error!` give an error type value. I think it more consistent than cause an exception immediately, consider that `make datatype! ...` will return a value of the datatype for all the other datatypes.
2. If you want to cause an exception, use `cause-error`, **do** the error value or if you prefer shorter code, write a helper function.
```
throw-error: func [msg][do make error! msg]
```

**Comment by bitbegin** [comment]
`do` is that what i want

**Comment by endo64** [comment]
@greggirwin What do you think adding `throw-error` to the language, it would be good for compatibility with Rebol as well.

**Comment by DideC** [comment]
OK, but for that to be clear to me: what about the `i256-to-int` function example above?
If I understand correctly the behaviour, the line with the `make error` is just useless actually, no ?

**Comment by endo64** [comment]
@DideC  It should be updated to `do make error! "i256-to-int error!"` I think.
